### ISSUE TITLE
Add a way to set the train/test mode for model explicitly for prediction and adversarial crafting

### ISSUE THREAD

**Issue by AkashGanesan** [IBM/adversarial-robustness-toolbox#28]
**Is your feature request related to a problem? Please describe.**
I'm trying to set dropout layers to be in train mode while in test mode.  Currently, predict will force the entire model into test mode.  

**Describe the solution you'd like**
I'll be using Pytorch models so if there is a way where I have to can set the model's learning phases explicitly instead of the predict forcing the model to true would be what I'm looking for essentially.

**Describe alternatives you've considered**
```
 def predict(self, x, logits=False, batch_size=128, force_test_mode=True):
    if force_test_mode: 
       self._model.train(False)
```

If this is needed for crafting adversarial examples is something I'm not very sure of.  I'll need to dig up some more literature before I can be certain if that'll be needed as well.

**Comment by ririnicolae** [comment]
This feature has been implemented for all backend frameworks. This is now part of the `Classifier` API. You can call `classifier.set_learning_phase()` for setting and the property `classifier.learning_phase` to check if a value has been set. https://github.com/IBM/adversarial-robustness-toolbox/commit/a1b750e2951b1894b44f1d598161daf122e6bd81 closes this issue. @AkashGanesan let us know if this does not fit your needs.
